front/src/components/Dialogs/TaskExportDialog.vue

```diff
@@ -104,32 +137,60 @@ export default defineComponent({
         )
       ).join('\n\n');
 
-      // download
       const blob = new Blob([template], {
         type: 'text/markdown',
       });
-      const link = document.createElement('a');
-      link.href = URL.createObjectURL(blob);
-      link.download = this.ctf.title + '.md';
-      link.click();
-      URL.revokeObjectURL(link.href);
+
+      this.downloadBlob(blob);
+    },
+
+    async exportTasksAsZip(tasks: Task[]) {
+      const zip = new JSZip.default();
+
+      const files = await Promise.all(
+        tasks.flatMap(async (task) => {
+          return {
+            title: task.title,
+            content: await this.downloadTaskMarkdown(task),
+          };
+        }, this)
+      );
+
+      zip.file(
+        `${this.ctf.title} - Overview.md`,
+        this.createIntroduction(tasks)
+      );
+
+      for (const f of files) {
+        zip.file(f.title + '.md', f.content);
```

Comment:
Maybe we can use the task slug here instead of title.
Just to avoid having unconveniant chars in the file name.

Reply:
When I put some Chinese characters as the task title, it is properly handled by JSZip. If I use a slug, it will result in the name 'no-slug-for-you.md'. If multiple tasks have such characters, they will overwrite each other in the zip because they all have the slug 'no-slug-for-you'. So I suggest to keep it like this since it is already properly handled by JSZip.